### DISCUSSION TITLE
Request payload validation

### DISCUSSION THREAD

**Discussion by sebastien-savalle** [mockoon/mockoon#1434] [Ideas]
Hi,

Mockoon is a very useful tool for prototyping and to run tests when we don't have access to remote system.
Sometimes, we have integration issues because we are validating requests have been correctly sent to mockoon and the response is valid. But, we are validating the request payload sent to mockoon.

I would like to ensure that the request body is valid, and optionally return a bad request when this is not compliant with a given schema.
I would be nice to integrate AJV for example. https://ajv.js.org/

What do you think ?
Regards,
Sebastien

**Comment by 255kb** [comment]
Hi, 

This is a growing topic I think and a great idea. 
A "quick" way to implement this could be to create a new rule type "JSON schema" that would validate any source (body, etc) against a data bucket containing a JSON schema. Like that we reuse the existing features and introduce minimal changes. 
Rules can also be used with wildcard routes and fallback mode to create global rules or guards (https://mockoon.com/docs/latest/route-responses/global-routes-with-rules/).
What do you think? 

Maybe you can create an issue, so we can put it on the roadmap? Thanks

**Comment by sebastien-savalle** [comment]
Thanks, I have created an issue.
